### DISCUSSION TITLE
Using turborepo without building shared libs (next-transpile-packages)

### DISCUSSION THREAD

**Discussion by izi-p** [vercel/turborepo#4206] [Help] [answered]
Hey,

We are trying to set up turborepo on our monorepo project.

We have this architecture:

```
package.json
packages/
---------- app1/
--------------- package.json
---------- app2/
--------------- package.json
---------- lib1/
--------------- package.json 
---------- lib2/
--------------- package.json
```
         

Apps are next.js application (v13) and libs are internal pacakges, pure react components libraries. Libs are not published, nor built in a standalone way.

app use code from libs :

_packages/app1/package.json_
```
"dependencies": {
   "lib1": "1.0.0",
   "lib2": "1.0.0",
}
```

and build them using transpilePackages :

_packages/app1/next.config.js_
```
transpilePackages: [
    'lib1'
    'lib2',
  ],
```

Here is our turbo.json 
```
{
  "$schema": "https://turbo.build/schema.json",
  "pipeline": {
    "build": {
      "inputs": [
        "src/**/*.tsx",
        "src/**/*.ts",
        "config/*.js",
        "package.json",
        "tsconfig.json",
        "next.config.js"
      ],
      "outputs": [".next/**"],
    },
  }
}

```

Turborepo doesn't understand to create a new build hash for app when lib1 or lib2 changes.
Does someonehave an idea how to implement that? Is it even possible ? 😢 

Thanks 🙏

**Comment by anthonyshew** [comment, marked as answer]
A few issues I see: 
- `packages/app1/package.json` should use the package syntax for your package manager.
```
"dependencies": {
   "lib1": "*", // or "workspace:*" for pnpm and yarn v3+
   "lib2": "*",// or "workspace:*" for pnpm and yarn v3+
}
```
- You likely do not need to use `inputs` at all.
- Your build needs to reflect it's topological dependencies. Example:
```
{
  "$schema": "https://turbo.build/schema.json",
  "pipeline": {
    "build": {
      "dependsOn": ["^build"],
      "outputs": [".next/**"],
    },
  }
}

```

You may benefit from reading this doc: https://turbo.build/repo/docs/getting-started/create-new
